core/camel-core-processor/src/main/java/org/apache/camel/processor/aggregate/AggregateProcessor.java

```diff
@@ -1877,4 +1882,44 @@ public int forceDiscardingOfAllGroups() {
         return total;
     }
 
+    /**
+     * Synchronization class to avoid busy-loop when waiting for exchanges to be processed during shutdown.
+     */
+    protected static final class WaitableInteger extends AbstractQueuedSynchronizer {
+
+        // await for this integer to be equal to zero
+        public void await() throws InterruptedException {
+            acquireSharedInterruptibly(1);
+        }
+
+        // decrement the integer
+        public void decrement() {
+            releaseShared(-1);
+        }
+
+        // increment the integer
+        public void increment() {
+            releaseShared(1);
+        }
+
+        // reset the integer to zero, this call won't trigger threads blocked on an await() call
+        public void reset() {
+            setState(0);
+        }
+
+        protected int tryAcquireShared(int acquires) {
```

Comment:
the `acquires` parameters seems to be unused, don't know if that's expected but is so a small comment could help understanding why.